File: tbsmith/json.php.ssl1304.com/controllers/word_service.php
<?php

include "models/DBConnector.php";

class WordService {

    var $db;

    function __construct() {
        $dbc = new DBConnector();
        $this->db = $dbc->connect();
    }

    function fnsearch($fn) {
        $sql = 'select * from address where firstname like ? limit 10';
        $q = $this->db->prepare($sql);
        $q->execute(array($fn));
        echo json_encode($q->fetchAll());
    }

    function duck($text='default') {

        $url = "http://api.duckduckgo.com/?q=" 
            . urlencode($text)
            . "&format=json&pretty=1";

        $contents = file_get_contents($url);

        $data = json_decode($contents);

        echo "<pre>";
        var_dump($data);
        echo "</pre>";
    }

}

?>

File: alyssamichelle/python/python/controllers/word_service.py
#!/usr/bin/python

import json
import urllib2
from models.DBConnector import DBConnector


class WordService():

    def __init__(self):
        dbcon = DBConnector()
        self.cnx = dbcon.get_connection()
        self.cursor = self.cnx.cursor()

    def fnsearch(self, fn):

        sql = ("SELECT firstname, lastname, city, state, "
               + "zip, phone FROM million WHERE firstname "
               + "LIKE '{}' LIMIT 10").format(fn)

        self.cursor.execute(sql)
        
        allDefs = []
        
        for firstname, lastname, city, state, zip, phone in self.cursor:
            
            allDefs.append({
                 'firstname': firstname,
                 'lastname': lastname,
                 'city': city,
                 'state': state,
                 'zip': zip,
                 'phone': phone})
        
        print json.dumps(allDefs)    

        self.cursor.close()
        self.cnx.close()

    def duck(self, text="default"):
        response = urllib2.urlopen(
            "http://api.duckduckgo.com/?q="
            + text + "&format=json&pretty=1")
        resp = response.read()
        gotJson = json.loads(resp)
        print gotJson
